JetNews/app/src/main/java/com/example/jetnews/ui/JetnewsApp.kt

```diff
@@ -54,42 +54,51 @@ fun JetnewsApp(
             val navigationActions = remember(navController) { JetnewsNavigationActions(navController) }
 
             val coroutineScope = rememberCoroutineScope()
-            // This top level scaffold contains the app drawer, which needs to be accessible
+            // This top level composable contains the app drawer, which needs to be accessible
             // from multiple screens. An event to open the drawer is passed down to each
             // screen that needs it.
-            val scaffoldState = rememberScaffoldState()
+            val drawerState = rememberDrawerState(DrawerValue.Closed)
 
             val navBackStackEntry by navController.currentBackStackEntryAsState()
             val currentRoute = navBackStackEntry?.destination?.route ?: JetnewsDestinations.HOME_ROUTE
 
             BoxWithConstraints {
                 val windowSize = getWindowSize(maxWidth)
-                Scaffold(
-                    scaffoldState = scaffoldState,
-                    // If the window size is Compact, show the AppDrawer.
-                    // Otherwise, a NavRail will be shown in individual screens
-                    drawerContent = if (windowSize == WindowSize.Compact) {
-                        {
-                            AppDrawer(
-                                currentRoute = currentRoute,
-                                navigateToHome = navigationActions.navigateToHome,
-                                navigateToInterests = navigationActions.navigateToInterests,
-                                closeDrawer = {
-                                    coroutineScope.launch { scaffoldState.drawerState.close() }
-                                }
-                            )
-                        }
-                    } else {
-                        null
-                    }
-                ) { innerPaddingModifier ->
+
+                val allowDrawerToBeShown = windowSize == WindowSize.Compact
+
+                // Determine the drawer state to pass to the modal drawer.
+                val sizeAwareDrawerState = if (allowDrawerToBeShown) {
```

Comment:
This looks like an implementation detail I might not want to see when looking at `JetnewsApp`. Should we move this logic to a separate private composable function that decides which drawer state to return?

```
@Composable
private fun rememberSizeAwareDrawerState(allowDrawerToBeShown: Boolean) {
    val drawerState = rememberDrawerState(DrawerValue.Closed)
    return if (allowDrawerToBeShown) ...
}
```

Reply:
Updated in the latest commit!